src/qibo/models/error_mitigation.py

```diff
@@ -592,8 +632,340 @@ def apply_randomized_readout_mitigation(
+
+        qubit_map (list, optional): the qubit map. If ``None``, a list of range of circuit's qubits is used. Defaults to ``None``.
+        noise_model (qibo.models.noise.Noise, optional): the noise model to be applied. Defaults to ``None``.
+        nshots (int, optional): the number of shots for the circuit execution. Defaults to :math:`10000`.
+        n_training_samples (int, optional): the number of training samples. Defaults to 10.
+        full_output (bool, optional): if ``True``, this function returns additional
+            information: ``val``, ``optimal_params``, ``train_val``. Defaults to ``False``.
+        backend (qibo.backends.abstract.Backend, optional): the backend to be used in the execution.
+            If None, it uses the global backend. Defaults to ``None``.
+
+    Returns:
+        mitigated_expectation (float): the mitigated expectated value.
+        mitigated_expectation_std (float): the standard deviation of the mitigated expectated value.
+        dep_param (float): the depolarizing parameter.
+        dep_param_std (float): the standard deviation of the depolarizing parameter.
+        lambda_list (list): the list of the depolarizing parameters.
+        data (dict): the data dictionary containing the noise-free and noisy expectation values obtained with the training circuits.
+
+    Reference:
+        1. Dayue Qin, Yanzhu Chen et al, *Error statistics and scalability of quantum error mitigation formulas*.
+           `arXiv:2112.06255 [quant-ph] <https://arxiv.org/abs/2112.06255>`_.
+    """
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if qubit_map is None:
+        qubit_map = list(range(circuit.nqubits))
+
+    training_circuits = [
+        error_sensitive_circuit(circuit, observable, backend=backend)[0]
+        for _ in range(n_training_samples)
+    ]
+
+    data = {"noise-free": [], "noisy": []}
+    lambda_list = []
+
+    for training_circuit in training_circuits:
+        circuit_result = training_circuit(nshots=nshots)
+        expectation = observable.expectation_from_samples(circuit_result.frequencies())
+
+        noisy_expectation = apply_problem_with_readout_conf(
+            training_circuit,
+            observable,
+            noise_model,
+            nshots,
+            readout,
+            qubit_map,
+            backend=backend,
+        )
+
+        data["noise-free"].append(expectation)
+        data["noisy"].append(noisy_expectation)
+        lambda_list.append(1 - noisy_expectation / expectation)
+
+    dep_param = np.mean(lambda_list)
+    dep_param_std = np.std(lambda_list)
+
+    noisy_expectation = apply_problem_with_readout_conf(
+        circuit, observable, noise_model, nshots, readout, qubit_map, backend=backend
+    )
+
```

Comment:
Maybe compute these separately to avoid multiple identical calculations
```suggestion
    1_dep_squared = (1 - dep_param) ** 2
    dep_std_squared = dep_param_std ** 2
```